### PR TITLE
test: add jest-dom matchers and eslint plugin

### PR DESCRIPTION
**Pull request by kretajak**
## Summary

Testing Library provides nice set of custom matchers that are more declarative and easier to understand. Along with `@testing-library/jest-dom` there is an accompanying `eslint-plugin-jest-dom` which enforce using these selectors. 

## Check List

- [x] `pnpm run prettier` for formatting code and docs

### CODE REVIEW

**Comment by kretajak** [comment]
Seems like I'm not be able to satisfy TS versions below `4.5.0` as they lacks support of `import type`. It's similar issue to https://github.com/testing-library/user-event/issues/1161.

**Comment by dai-shi** [comment]
Can you install an old version for old TS?

https://github.com/pmndrs/jotai/blob/71cbc2f2ed3c6968a2b024fb77a1f6756e67444e/.github/workflows/test-old-typescript.yml#L70

File: tests/setup.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+import '@testing-library/jest-dom/vitest'
```
Comment (dai-shi) [inline tests/setup.ts:1]: I use the file name `vitest-setup.js` elsewhere. Maybe `vitest-setup.ts` here?

https://github.com/jotaijs/jotai-zustand/blob/main/tests/vitest-setup.js

**Comment by dai-shi** [comment]
- https://github.com/pmndrs/zustand/pull/2850
- https://github.com/pmndrs/valtio/pull/997

File: .github/workflows/test-old-typescript.yml [hunk 2]
```diff
@@ -85,6 +85,10 @@ jobs:
         if: ${{ matrix.typescript == '4.4.4' || matrix.typescript == '4.3.5' || matrix.typescript == '4.2.3' || matrix.typescript == '4.1.5' ||  matrix.typescript == '4.0.5' || startsWith(matrix.typescript, '3.') }}
         run: |
           pnpm add -D vitest@0.33.0 @vitest/coverage-v8@0.33.0 @vitest/ui@0.33.0
+          pnpm add -D @testing-library/jest-dom@5 @types/testing-library__jest-dom@5 
+          pnpm add -D @types/jest@27.4.1
+          sed -i~ 's/"@testing-library\/jest-dom"/"@types\/testing-library__jest-dom"/' tsconfig.json
+
```
Comment (dai-shi) [inline .github/workflows/test-old-typescript.yml:91]:
```suggestion
```

**Review by dai-shi — APPROVED** [review]
LGTM

File: tests/setup.ts (hunk 1 above)
Comment (kretajak) [reply tests/setup.ts:1]: Oh. I was about to change that, but look like PR is merged. :)

Comment (dai-shi) [reply tests/setup.ts:1]: I wasn't sure if you were going to change. Feel free to send new PRs.

**Pull request merged**